lib/puppet/util/windows/service.rb

```diff
@@ -377,13 +380,16 @@ def service_state(service_name)
     # @param [:string] service_name name of the service to query
     # @return [QUERY_SERVICE_CONFIGW.struct] the configuration of the service
     def service_start_type(service_name)
-      config = nil
+      start_type = nil
+      start_type_value = nil
       open_service(service_name, SC_MANAGER_CONNECT, SERVICE_QUERY_CONFIG) do |service|
-        config = query_config(service)
+        query_config(service) do |config|
+          start_type_value = config[:dwStartType]
+          start_type = SERVICE_START_TYPES[start_type_value]
```

Comment:
Same case here -- probably better to write it as `SERVICE_START_TYPES[config[:dwStartType]]`.